### PR TITLE
adds scheduler to clean db

### PR DESCRIPTION
**Pull request by tsop14**
Currently, when a recipe is opened its status is set to 'Pending'. If the recipe is skipped (currently this just refreshes the page), or the browser window closed the status remains on 'Pending'.

- Sets all pending recipes' status to 'New' at midnight to allow them to be curated again.

### CODE REVIEW

File: ui/app/schedule/DBCleanSheduler.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import org.quartz.{ Job, JobDataMap, JobExecutionContext }
+import org.quartz.impl.StdSchedulerFactory
+
+class DBCleaner(db: DB) {
+
+  private val scheduler = StdSchedulerFactory.getDefaultScheduler()
+
+  private def buildJobDataMap: JobDataMap = {
+    val map = new JobDataMap()
+    map.put("DB", db.resetStatus)
```
Comment (LATaylor-guardian) [inline ui/app/schedule/DBCleanSheduler.scala]: Can this not be passed into a constructor?

File: ui/app/schedule/DBCleanSheduler.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    map
+  }
+
+  val jobDetail = newJob(classOf[HouseKeepingJob])
```
Comment (LATaylor-guardian) [inline ui/app/schedule/DBCleanSheduler.scala]: This can be private.

File: ui/app/schedule/DBCleanSheduler.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    .withIdentity("cleanDB")
+    .usingJobData(buildJobDataMap)
+    .build()
+
+  val trigger = newTrigger()
```
Comment (LATaylor-guardian) [inline ui/app/schedule/DBCleanSheduler.scala]: This can be private.

File: ui/app/com/gu/recipeasy/AppComponents.scala [hunk 4]
```diff
@@ -29,6 +30,7 @@ class AppComponents(context: Context)
   applicationLifecycle.addStopHook(() => Future.successful(dbContext.close()))
   val db = new DB(dbContext)
   val messagesApi: MessagesApi = new DefaultMessagesApi(environment, configuration, new DefaultLangs(configuration))
+  val dbScheduler = new DBCleaner(db)
```
Comment (LATaylor-guardian) [inline ui/app/com/gu/recipeasy/AppComponents.scala]: Should this be called `dbCleaner`?

Comment (tsop14) [inline ui/app/com/gu/recipeasy/AppComponents.scala]: I think it should have scheduler in its name? I can change the class name.

File: ui/app/schedule/DBCleanSheduler.scala (hunk 1 above)
Comment (tsop14) [inline ui/app/schedule/DBCleanSheduler.scala]: No :(

File: ui/app/schedule/DBCleanSheduler.scala (hunk 3 above)
Comment (tsop14) [inline ui/app/schedule/DBCleanSheduler.scala]: Changed

**Comment by cb372** [comment]
👍

**Pull request merged**